drivers/modem/Kconfig

```diff
@@ -30,15 +30,66 @@ config MODEM_RECEIVER
 	  in that callbacks are executed in a different work queue and data is
 	  passed around in k_pipe structures.
 
+if MODEM_RECEIVER
+
 config MODEM_RECEIVER_MAX_CONTEXTS
 	int "Maximum number of modem receiver contexts"
-	depends on MODEM_RECEIVER
 	range 1 10
 	default 1
 	help
 	  Maximum number of modem receiver contexts to handle.  For most
 	  purposes this should stay at 1.
 
+endif # MODEM_RECEIVER
+
+config MODEM_CONTEXT
+	bool "Enable modem context helper driver"
+	help
+	  This driver allows modem drivers to communicate with an interface
+	  using custom defined protocols. Driver doesn't inspect received data
+	  and all aspects of received protocol data are handled by application
+	  work method provided.  This driver combines abstractions for:
+	  modem interface, command handler, pin config and socket handling.
+
+if MODEM_CONTEXT
+
+config MODEM_CONTEXT_MAX_NUM
+	int "Maximum number of modem contexts"
+	default 1
+	help
+	  Maximum number of modem contexts to handle.  For most
+	  purposes this should stay at 1.
+
+config MODEM_CONTEXT_VERBOSE_DEBUG
+	bool "Enable verbose debug in modem context"
```

Comment:
Could skip the "Enable ..." parts here and below maybe, since it's clear in the `menuconfig` from the checkbox.

Something like

    bool "Verbose debug output in modem context"
    bool "UART-based modem interface"
    bool "Generic modem command handler"
    bool "u-blox SARA modem driver"
    ...

etc.